### ISSUE TITLE
Problem with jmespath: unexpected behavior with single items (not arrays)

### ISSUE THREAD

**Issue by colleenXu** [biothings/biothings.api#323]
I'm trying to query MyChem's drugcentral info and only keep `drugcentral.bioactivity` objects with the matching action_type value (POSITIVE MODULATOR). So I'm setting the parameter `jmespath` to `drugcentral.bioactivity|[?action_type=='POSITIVE MODULATOR']`

But if I try using that jmespath on this document [C0017845_doesnt_work.json](https://github.com/biothings/biothings.api/files/14924033/C0017845_doesnt_work.json), I end up with `drugcentral.bioactivity: null` which is unexpected. (To retrieve this document yourself, use [this GET query](https://mychem.info/v1/query?q=drugcentral.xrefs.umlscui:C0017845&fields=drugcentral)))

<details><summary>POST-query with chemical C0017845 that returns null</summary>
<p>

```
curl --location --globoff 'https://mychem.info/v1/query?size=1000&fields=drugcentral.bioactivity.uniprot.uniprot_id,drugcentral.bioactivity.action_type,drugcentral.bioactivity.act_source,drugcentral.bioactivity.organism&jmespath=drugcentral.bioactivity|[?action_type=='POSITIVE%20MODULATOR']' \
--header 'Content-Type: application/json' \
--data '{
  "q": [
      [ "C0017845", "POSITIVE MODULATOR"]
      ],
  "scopes": ["drugcentral.xrefs.umlscui", "drugcentral.bioactivity.action_type"]
}'
```

</p>
</details> 

---

VS when I try using jmespath on a different document [C0018549_works.json](https://github.com/biothings/biothings.api/files/14924074/C0018549_works.json), it works as-intended. (To retrieve this document yourself, use [this GET query](https://mychem.info/v1/query?q=drugcentral.xrefs.umlscui:C0018549&fields=drugcentral)))

<details><summary>A POST query with chemical C0018549 that works as-intended</summary>
<p>

```
curl --location --globoff 'https://mychem.info/v1/query?size=1000&fields=drugcentral.bioactivity.uniprot.uniprot_id,drugcentral.bioactivity.action_type,drugcentral.bioactivity.act_source,drugcentral.bioactivity.organism&jmespath=drugcentral.bioactivity|[?action_type=='POSITIVE MODULATOR']' \
--header 'Content-Type: application/json' \
--data '{
  "q": [
      [ "C0018549", "POSITIVE MODULATOR"]
      ],
  "scopes": ["drugcentral.xrefs.umlscui", "drugcentral.bioactivity.action_type"]
}'
```

</p>
</details> 

I do notice a different between these two documents that may account for this:
* in the first case, drugcentral.bioactivity is an object (since there was only 1 thing)
* in the second case, drugcentral.bioactivity is an array of objects 

If this is the key, I'd like jmespath to be able to gracefully handle both situations...

**Comment by colleenXu** [comment]
Another example is described in https://github.com/biothings/biothings_explorer/issues/316#issuecomment-2030832144

**Comment by colleenXu** [comment]
I've updated the opening post to match the [discussions](https://suwulab.slack.com/archives/CC19LHAF2/p1712264667111049?thread_ts=1712018348.531719&cid=CC19LHAF2) in Slack/add the json documents. 

The proposed fix is to adjust the always_list parameter behavior (so it works before jmespath) and then add it to ALL queries using jmespath, just in case.

**Comment by colleenXu** [comment]
Assigning to myself to review and see if this is fixed.

**Comment by colleenXu** [comment]
Using always_list fixes this behavior! Going to add to all jmespath operations while doing https://github.com/biothings/biothings_explorer/issues/733...so track updates there.

**Comment by colleenXu** [comment]
FYI: 
* I think the key BioThings change is here https://github.com/biothings/biothings.api/commit/7df44a3fe7e9e0d76a54c76b22353ed34d07de70. I think this allows the `always_list` transformation to be done first, before jmespath processing is done ([see convo in lab Slack](https://suwulab.slack.com/archives/CC19LHAF2/p1712264667111049?thread_ts=1712018348.531719&cid=CC19LHAF2)). 
* MyChem was updated with that BioThings change May 9 [(lab Slack link)](https://suwulab.slack.com/archives/CC19LHAF2/p1715277356207489)
* **I don't know if other APIs have been updated.**

**Comment by colleenXu** [comment]
Regarding current x-bte operations:
* I ended up adding `always_list` to all MyChem `drugcentral.bioactivity` x-bte operations: they all use jmespath and the field's value could be an object (single) or array (multiple)
* I ended up NOT adding `always_list` to any other current use of jmespath. For those cases and fields, I saw single-element arrays. So I assumed the field's value was always an array. See comments in [these](https://github.com/NCATS-Tangerine/translator-api-registry/commit/30e0a8ac2e5152eb757e014dc446b9199bcde49f) [commits](https://github.com/NCATS-Tangerine/translator-api-registry/commit/57d6b019f2ae32402038b2599f8ac99bce2745c2).
